### ISSUE TITLE
[Question] Can we customize the visual encoder and llm used? For example, replace llama with opt？

### ISSUE THREAD

**Issue by Akiraxty** [haotian-liu/LLaVA#208]

**Comment by haotian-liu** [comment]
Yes you can.  We have largely refactored our code base to make it easier to swap out the visual encoder / LLMs. 

Checkout our initial guide in replacing the LLMs: https://github.com/haotian-liu/LLaVA/blob/main/docs/Customize_Component.md
